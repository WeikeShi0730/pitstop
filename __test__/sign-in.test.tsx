import { render, RenderResult, fireEvent } from "@testing-library/react";
import SignIn from "../components/sign-in.component";
import createMockRouter from "../test-utils/createMockRouter";
import { RouterContext } from "next/dist/shared/lib/router-context";

let documentBody: RenderResult;
describe("<SignIn />", () => {
  it("Test <SignIn />", () => {
    documentBody = render(<SignIn />);
    expect(
      documentBody.getByText("Alrady have an account? Sign in üîê")
    ).toBeInTheDocument();
    expect(
      documentBody.getAllByRole("button", { name: /Sign in/i })
    ).toHaveLength(2);

    const { baseElement } = documentBody;
    expect(baseElement).toMatchSnapshot();
  });
  it("Test click router <SignIn />", () => {
    const router = createMockRouter({});
    documentBody = render(
      <RouterContext.Provider value={router}>
        <SignIn />
      </RouterContext.Provider>
    );
    fireEvent.click(
      documentBody.getByRole("button", { name: /Forgot password?/i })
    );
    expect(router.push).toHaveBeenCalledWith("/reset-password");
  });
});
