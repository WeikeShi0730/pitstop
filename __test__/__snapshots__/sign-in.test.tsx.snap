// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignIn /> Test <SignIn /> 1`] = `
<body>
  <div>
    <div
      class="max-w-xs sm:max-w-sm md:max-w-md lg:max-w-lg m-auto text-slate-700 bg-slate-100 bg-opacity-30 rounded-lg py-8 px-10 shadow-md hover:shadow-slate-500 transition-all ease-in-out duration-200"
    >
      <div
        class="text-base lg:text-xl font-light mt-4 mb-12 text-center"
      >
        Alrady have an account? Sign in üîê
      </div>
      <form>
        <div
          class="my-5"
        >
          <label
            class="relative text-sm md:text-base"
          >
            <input
              aria-label="signInEmail"
              class="w-full p-2 border-b border-slate-700 outline-none bg-transparent mb-4 rounded-none transition duration-200"
              id="email"
              name="email"
              placeholder=" "
              required=""
              type="email"
            />
            <span
              class="absolute left-0 top-0 px-2 transition duration-200 input-text"
            >
              Email
            </span>
          </label>
        </div>
        <div
          class="my-5"
        >
          <label
            class="relative text-sm md:text-base"
          >
            <input
              class="w-full p-2 border-b border-slate-700 outline-none bg-transparent mb-4 rounded-none transition duration-200"
              id="signInPassword"
              name="password"
              placeholder=" "
              required=""
              type="password"
            />
            <span
              class="absolute left-0 top-0 px-2 transition duration-200 input-text"
            >
              Password
            </span>
          </label>
        </div>
        <div
          class="flex justify-center items-center mt-6"
        >
          <button
            class="text-sm md:text-base bg-slate-700 py-2 px-4 text-slate-100 rounded outline-none font-light hover:shadow-md hover:shadow-slate-700 hover:bg-slate-500 hover:text-white transition-all ease-in-out duration-200"
            id="signInButton"
            type="submit"
          >
            Sign in
          </button>
        </div>
      </form>
      <div
        class="w-full max-w-md mx-auto mt-4"
      >
        <div
          class="flex justify-center"
        >
          <div>
            <button
              class="text-sm md:text-base bg-blue-500 py-2 px-4 text-slate-100 rounded outline-none font-light hover:shadow-md hover:shadow-blue-500 hover:bg-blue-400 hover:text-white transition-all ease-in-out duration-200"
            >
              Sign in with Google
            </button>
          </div>
        </div>
      </div>
      <div
        class=""
      >
        <button
          class="text-sm md:text-base mt-4 font-light hover:text-slate-500"
        >
          Forgot password?
        </button>
      </div>
    </div>
  </div>
</body>
`;
